src/analytics/constants.ts

```diff
@@ -0,0 +1,138 @@
+  ALL_WALLET_ADDRESSES_CONNECTED = 'all_wallet_addresses_connected',
+  ALL_WALLET_CHAIN_IDS = 'all_wallet_chain_ids',
+  USER_AGENT = 'user_agent',
+  BROWSER = 'browser',
+  DARK_MODE = 'is_dark_mode',
+  EXPERT_MODE = 'is_expert_mode',
+  SCREEN_RESOLUTION_HEIGHT = 'screen_resolution_height',
+  SCREEN_RESOLUTION_WIDTH = 'screen_resolution_width',
+  WALLET_ADDRESS = 'wallet_address',
+  WALLET_TYPE = 'wallet_type',
+}
+
+export enum BROWSER {
+  FIREFOX = 'Mozilla Firefox',
+  SAMSUNG = 'Samsung Internet',
+  OPERA = 'Opera',
+  INTERNET_EXPLORER = 'Microsoft Internet Explorer',
+  EDGE = 'Microsoft Edge (Legacy)',
+  EDGE_CHROMIUM = 'Microsoft Edge (Chromium)',
+  CHROME = 'Google Chrome or Chromium',
+  SAFARI = 'Apple Safari',
+  BRAVE = 'Brave',
+  UNKNOWN = 'unknown',
+}
+
+export enum WALLET_CONNECTION_RESULT {
+  SUCCEEDED = 'Succeeded',
+  FAILED = 'Failed',
+}
+
+export enum SWAP_PRICE_UPDATE_USER_RESPONSE {
+  ACCEPTED = 'Accepted',
+  REJECTED = 'Rejected',
+}
+
+/**
+ * Known pages in the app. Highest order context.
+ */
+export enum PageName {
+  TOKEN_DETAILS_PAGE = 'token-details',
+  TOKENS_PAGE = 'tokens-page',
+  POOL_PAGE = 'pool-page',
+  SWAP_PAGE = 'swap-page',
+  VOTE_PAGE = 'vote-page',
+  // alphabetize additional page names.
+}
+
+/**
+ * Sections. Disambiguates low-level elements that may share a name.
+ * eg a `back` button in a modal will have the same `element`,
+ * but a different `section`.
+ */
+export enum SectionName {
+  CURRENCY_INPUT_PANEL = 'swap-currency-input',
+  CURRENCY_OUTPUT_PANEL = 'swap-currency-output',
+  WIDGET = 'widget',
+  // alphabetize additional section names.
+}
+
+/** Known modals for analytics purposes. */
+export enum ModalName {
+  CONFIRM_SWAP = 'confirm-swap-modal',
+  TOKEN_SELECTOR = 'token-selector-modal',
+  // alphabetize additional modal names.
+}
+
+/**
+ * Known element names for analytics purposes.
+ * Use to identify low-level components given a TraceContext
+ */
+export enum ElementName {
+  AUTOROUTER_VISUALIZATION_ROW = 'expandable-autorouter-visualization-row',
+  COMMON_BASES_CURRENCY_BUTTON = 'common-bases-currency-button',
+  CONFIRM_SWAP_BUTTON = 'confirm-swap-or-send',
+  CONNECT_WALLET_BUTTON = 'connect-wallet-button',
+  EXPLORE_BANNER = 'explore-banner',
+  EXPLORE_SEARCH_INPUT = 'explore_search_input',
+  IMPORT_TOKEN_BUTTON = 'import-token-button',
+  MAX_TOKEN_AMOUNT_BUTTON = 'max-token-amount-button',
+  NAVBAR_SEARCH_INPUT = 'navbar-search-input',
+  PRICE_UPDATE_ACCEPT_BUTTON = 'price-update-accept-button',
+  SWAP_BUTTON = 'swap-button',
+  SWAP_DETAILS_DROPDOWN = 'swap-details-dropdown',
+  SWAP_TOKENS_REVERSE_ARROW_BUTTON = 'swap-tokens-reverse-arrow-button',
+  TOKEN_SELECTOR_ROW = 'token-selector-row',
+  WALLET_TYPE_OPTION = 'wallet-type-option',
+  // alphabetize additional element names.
+}
+
+/**
+ * Known events that trigger callbacks.
+ * @example
+ *  <TraceEvent events={[Event.onClick]} element={name}>
+ */
+export enum Event {
```

Comment:
naming: Given these are not analytics events but browser events, could we update to `BrowserEvent`?

Reply:
Makes sense, however, this would not be backward compatible tho 🤔 Keep in mind the "interface" of the pkg has already been implemented in the front end.

Reply:
resolved offline!